kivy/core/window/window_sdl2.py

```diff
@@ -678,9 +689,14 @@ def mainloop(self):
                 self._focus = False
 
             elif action == 'windowenter':
-                self.dispatch('on_cursor_enter')
+                x, y = self._win.get_nearest_mouse_pos()
```

Comment:
`windowenter` is not proceeded by any other mouse events, right? And is it specific to mouse?

Reply:
From sdl doc:
```
SDL_WINDOWEVENT_ENTER - window has gained mouse focus
SDL_WINDOWEVENT_LEAVE - window has lost mouse focus
```

SDL reports the following events:

On Windows if app starts with cursor on the window:
- `windowexposed` is received
- no `windowenter`
- if mouse moves, `mousemotion` is received, `mouse_pos` is updated
- if any of mouse buttons is pressed, `mousebuttondown` is received, `mouse_pos` is updated, `on_mouse_down` is dispatched

If apps starts with cursor outside of window:
- `windowexposed` is received
- if cursor is moved on window, `windowenter` is received
- `mousemotion` is received, `mouse_pos` is updated

On Linux if app starts with cursor on the window:
- `windowmoved` is received - window is centered on the desktop
- `windowexposed` is received
- `windowenter` is received
- `windowfocusgained` is received
- if mouse moves, `mousemotion` is received, `mouse_pos` is updated
- if any of mouse buttons is pressed, `mousebuttondown` is received, `mouse_pos` is updated, `on_mouse_down` is dispatched

If app starts with cursor outside of the window:
- `windowmoved` is received
- `windowexposed` is received
- `windowfocusgained` is received
- if cursor is moved on window, `windowenter` is received
- `mousemotion` is received, `mouse_pos` is updated


After this on both platforms, moving cursor outside of the window (or minimizing it) will receive `windowleave` and moving cursor on the window will receive `windowenter`. 

For drop text/file case, on Linux we have `windowenter` and then drop events, on Windows we have drop events and then `windowenter` and `mousemotion`.